runelite-client/src/main/java/net/runelite/client/plugins/itemstats/potions/StaminaPotion.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2016-2018, Adam <Adam@sigterm.info>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.itemstats.potions;
+
+import net.runelite.api.Client;
+import net.runelite.api.EquipmentInventorySlot;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import net.runelite.api.ItemContainer;
+import net.runelite.api.ItemID;
+import net.runelite.client.plugins.itemstats.StatBoost;
+import static net.runelite.client.plugins.itemstats.stats.Stats.RUN_ENERGY;
+
+public class StaminaPotion extends StatBoost
+{
+	public StaminaPotion()
+	{
+		super(RUN_ENERGY, false);
+	}
+
+	private static final int RING_SLOT = EquipmentInventorySlot.RING.getSlotIdx();
+
+	@Override
+	public int heals(Client client)
+	{
+		boolean hasEquippedChargedRingOfEndurance = false;
+
+		ItemContainer equipContainer = client.getItemContainer(InventoryID.EQUIPMENT);
+		if (equipContainer != null)
+		{
+			Item ring = equipContainer.getItem(RING_SLOT);
+			hasEquippedChargedRingOfEndurance = ring != null && ring.getId() == ItemID.RING_OF_ENDURANCE;
```

Comment:
`hasEquippedChargedRingOfEndurance` is not needed. Instead, you can simply `return 40;` here if the condition is matched, and `return 20;` as the base case for this function, eg.

```java
@Override
public int heals(Client client)
{
	final ItemContainer equipment = client.getItemContainer(InventoryID.EQUIPMENT);
	if (equipment != null)
	{
		final Item ring = equipment.getItem(EquipmentInventorySlot.RING.getSlotIdx());
		if (ring != null && ring.getId() == ItemID.RING_OF_ENDURANCE)
		{
			return 40;
		}
	}
	return 20;
}
```